mgo_test.go

```diff
@@ -0,0 +1,41 @@
+package storage
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type checkCollection struct {
+	t      *testing.T
+	filter bson.D
+	value  interface{}
+	check  bool
+}
+
+func (c *checkCollection) ReplaceOne(ctx context.Context, filter interface{}, replacement interface{}, opts ...*options.ReplaceOptions) (*mongo.UpdateResult, error) {
+	assert.Equal(c.t, c.filter, filter)
+	assert.Equal(c.t, c.value, replacement)
+	c.check = true
+	return &mongo.UpdateResult{}, nil
+}
+
+func (c *checkCollection) calledReplaceOne() bool {
+	return c.check
+}
+
+func TestClient_Store(t *testing.T) {
```

Comment:
Favor adicionar um outro teste para pegar erros. Por exemplo, quando o método Store for chamado antes do connect.